index.js

```diff
@@ -176,20 +334,33 @@ async function loadModels(fastify) {
       } catch (error) {
         throw new Error('Failed to create view', { cause: error })
       }
-    } else {
-      await createIndexes(collection, indexes || [], PREFIX_OF_INDEX_NAMES_TO_PRESERVE)
+    }
+
+    let viewDependencies = {}
+    if (viewHasLookupEndpints) {
+      const sourceCollection = clone(mergedCollections.find(mod => mod.name === viewSourceCollectionName))
```

Comment:
Should we perform this change earlier when starting the creation of the view and document the requirement of defining the source collection among the crud collection definitions before creating a view is allowed?